pkg/operator/ceph/cluster/mon/spec.go

```diff
@@ -43,7 +43,7 @@ const (
 func (c *Cluster) getLabels(monConfig *monConfig, canary, includeNewLabels bool) map[string]string {
 	// Mons have a service for each mon, so the additional pod data is relevant for its services
 	// Use pod labels to keep "mon: id" for legacy
-	labels := controller.CephDaemonAppLabels(AppName, c.Namespace, "mon", monConfig.DaemonName, includeNewLabels)
+	labels := controller.CephDaemonAppLabels(AppName, c.Namespace, config.MonType, monConfig.DaemonName, c.ClusterInfo.NamespacedName().Name, "cephclusters.ceph.rook.io", includeNewLabels)
```

Comment:
c.ClusterInfo.NamespacedName().Name here updates to `testing` first and then it updates to `rook-ceph` the cluster name.

Reply:
Strange, I don't see where "testing" comes from. In a real cluster, `c.ClusterInfo.NamespacedName().Name` should only be set to the real name of the CephCluster CR. You see that in a real cluster? Does it show up on the mon canary pods, but not the mon daemon pods? Or which mon pods exactly do you see it on?

Reply:
Yes, the actual mons were updated with `testing`.

Actually, everything starts with `AdminClusterInfo` and which is named as testing, https://github.com/rook/rook/blob/d62abfd98d16d352e485556c0ee452abb761d8d3/pkg/daemon/ceph/client/info.go#L93
So calling the namespace-name is not possible directly from the cluster details, so need to set the name explicitly. 
With a lot of testing and de-bugging, I opened a fix for it https://github.com/rook/rook/pull/9203.